pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -577,11 +577,101 @@ func (r *Reconciler) processTargetPools(desired bool, poolFunc poolProcessor) er
 	return nil
 }
 
+func (r *Reconciler) ensureInstanceGroup(instanceGroupName string) error {
```

Comment:
Please add a godoc to explain what this function is doing

Reply:
Done